revolv/payments/tests/models/test_payment.py

```diff
@@ -1,101 +1,270 @@
-        self.assertEquals(Payment.objects.payments(user2).count(), 2)
+        self.assertEquals(Payment.objects.payments(user1).count(), 2)
+        self.assertEquals(Payment.objects.payments(user2).count(), 1)
 
     def test_donations(self):
-        """Test that we can bookkeep organic donation information."""
+        """
+        Test that we can bookkeep organic donation information.
+        """
         user1, user2 = RevolvUserProfile.factories.base.create_batch(2)
         project2 = Project.factories.base.create()
 
         self._create_payment(user1).save()
-        self._create_payment(user1, instrument_type=self.reinvestment).save()
-        self._create_payment(user1, instrument_type=self.repayment).save()
-
-        self._create_payment(user2, instrument_type=self.repayment).save()
+        self._create_payment(user1, payment_type=self.reinvestment).save()
 
         self.assertEquals(Payment.objects.donations(user1).count(), 1)
         self.assertEquals(Payment.objects.donations(user1, project2).count(), 0)
 
-        self._create_payment(user1, project2).save()
+        self._create_payment(user1, project=project2).save()
         self.assertEquals(Payment.objects.donations(user1, project2).count(), 1)
 
         self.assertEquals(Payment.objects.donations(user2).count(), 0)
 
-    def test_reinvestments(self):
-        """Test that we can bookkeep reinvestment information."""
+    def test_proportion_donated(self):
+        """
+        Verify that donation proportions are calculated correctly.
+        For user U, project P the proportion that a user has donated is:
+
+            Sum of U's donations to P / Total organic donation amount to P
+        """
+        user1, user2 = RevolvUserProfile.factories.base.create_batch(2)
+        admin1 = RevolvUserProfile.factories.admin.create()
+        project = Project.factories.base.create()
+
+        self.assertEquals(Payment.objects.donations(project=project).count(), 0)
+
+        self._create_payment(user1, 30.00, project).save()
+        self.assertEquals(Payment.objects.donations(project=project).count(), 1)
+        self.assertEquals(project.proportion_donated(user1), 1.0)
+
+        self._create_payment(user2, 10.00, project).save()
+        self.assertEquals(Payment.objects.donations(project=project).count(), 2)
+        self.assertEquals(project.proportion_donated(user2), 10.0 / 40.0)
+        self.assertEquals(project.proportion_donated(user1), 30.0 / 40.0)
+
+        project.complete_project()  # must complete project before repayment
+
+        self._create_admin_repayment(admin1, 100.00, project).save()
+        self._create_admin_reinvestment(admin1, 100.00, project).save()
+        self.assertEquals(Payment.objects.donations(project=project).count(), 2)
+        self.assertEquals(project.proportion_donated(user2), 10.0 / 40.0)
+        self.assertEquals(project.proportion_donated(user1), 30.0 / 40.0)
+
+    def test_bad_repayment(self):
```

Comment:
Great test, shouldn't we have one for reinvestments also that asserts that NotEnoughFundingException is thrown in the right places?